### PR TITLE
LPS-99846 Add analytics events

### PR DESCRIPTION
**Pull request by interaminense**

### CODE REVIEW

**Comment by natocesarrego** [comment]
ci:test:sf

**Comment by natocesarrego** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ const _handleFieldEdited = function(properties) {
 const _handleFieldBlurred = function(properties) {
 	const {fieldInstance} = properties;
 	const {pages} = this;
+	const now = new Date();
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js]: @interaminense, can we change this variable name to `dateNow` or `currentDate`? I think it'll be better for the developers understanding. What do you think about it?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js [hunk 2]
```diff
@@ -62,16 +63,31 @@ const _handleFieldBlurred = function(properties) {
...
+	this.fieldFocusDate = new Date();
+
 	handleFieldFocused(pages, properties).then(focusedFieldPages => {
 		this.setState({
 			pages: focusedFieldPages
 		});
 	});
+
+	Liferay.fire('ddmFieldFocus', {
+		fieldName: properties.fieldInstance.fieldName,
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js]: @interaminense, why did you need to use `properties.fieldInstance.fieldName` instead of `fieldInstance.fieldName`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js [hunk 3]
```diff
@@ -147,10 +169,23 @@ export default Component => {
...
+
 		getFormNode() {
 			return dom.closest(this.element, 'form');
 		}
 
+		getFormTitle() {
+			const form = this.getFormNode();
+			const formName = form && form.querySelector('.ddm-form-name');
+
+			return formName && formName.textContent;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js]: @interaminense, is there any chance of it return a `false` value instead of `undefined`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp [hunk 4]
```diff
@@ -197,7 +197,7 @@ long formInstanceId = ddmFormDisplayContext.getFormInstanceId();
 						Liferay.fire(
 							'ddmFormView',
 							{
-								formId: <%= formInstanceId %>,
+								formId: '<%= formInstanceId %>',
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp]: @interaminense, do you need to get this value as a `String` from now on?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js (hunk 1 above)
Comment (interaminense) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js]: Yes, @natocesarrego! Done: https://github.com/natocesarrego/liferay-portal/pull/177/commits/c961672db6885e1156b018b5a21cc12abec973d5

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js (hunk 2 above)
Comment (interaminense) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js]: **fieldInstance** is a properties property, but to keep the same pattern as **handleFieldFocused**, I destructured the properties property to access **fieldInstance** directly: https://github.com/natocesarrego/liferay-portal/pull/177/commits/81d8c799b8bb862c78256d8e920b77d409027c91

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js (hunk 3 above)
Comment (interaminense) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js]: Yes, it was possible to return `undefined`. But since I used this **getFormTitle** just to trigger the **formViewed** event, this event will no longer be fired in this withStore class, it will continue to fire in the `view.jsp` file, there is already the title. So I am removing this method in this commit: https://github.com/natocesarrego/liferay-portal/pull/177/commits/8e6f577daa6f357cdf693dcae56e444f0603d45f

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp (hunk 4 above)
Comment (interaminense) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp]: Yes, this is a request from the backend to receive IDs in `String` format.

**Comment by natocesarrego** [comment]
Thanks for your answers and adjustments, @interaminense. 😉 I'll forward your pull soon.

**Comment by interaminense** [comment]
Thanks, @natocesarrego

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by natocesarrego** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/79284. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
